SDK-1865: Add documentation how Spryker Code Upgrader works (#1726)
spryker/spryker-docs: docs/scu/dev/how-the-spryker-code-upgrader-works.md

---
title: How the Spryker Code Upgrader works
description: Spryker Code Upgrader overview
template: concept-topic-template
redirect_from:
  - /docs/paas-plus/dev/how-the-spryker-code-upgrader-works.html
---

To update a project, the Spryker Code Upgrader runs the following steps.

### 1. Identifies the Spryker modules to update

The Spryker Code Upgrader performs the following sub-steps:

> 1. To identify the modules to be updated, compares the information in the `composer.json` and `composer.lock` files with our latest released code.

Reviewer: ```suggestion
1. To identify the modules to be updated, it will compare the information present in the `composer.json` and `composer.lock` files with our latest released code.
```

> 2. Creates a list of modules and third-party libraries to be updated. Groups the modules according to how we released them.

Reviewer: ```suggestion
2. It creates a list of modules and third-party libraries to be updated. Groups the modules according to how we released them.
```

{% endinfo_block %}

### 2. Updates the modules and libraries

Using `composer`, the Upgrader updates the modules in groups. After updating the modules, the Upgrader returns the list of updated modules and proceeds to the next step.

If the Upgrader can’t update a module, it skips the module’s and the remaining groups. If all the groups failed to update, the Upgrader returns the errors and stops. With at least one group updated, it returns the list of updated modules and proceeds to the next step.

By default Spryker Code Upgrader updates only minor and patch versions. When Upgrader finds the group with [major release](/docs/scos/dev/architecture/module-api/semantic-versioning-major-vs.-minor-vs.-patch-release.html#what-is-a-major-release), it won't update it and informs you about that:

```bash
There is a major release available for module spryker/merchant-product-approval. 
Please follow the link below to find all documentation needed to help you upgrade to the latest release 
https://api.release.spryker.com/release-group/XXXX
```

> To continue the usage of Upgrader, install the major manually, and re-run the Upgrader.

Reviewer: ```suggestion
To continue the usage of the Upgrader, install the major update manually, and re-run the Upgrader.
```
